astropy/units/physical.py

```diff
@@ -357,22 +356,15 @@ def __eq__(self, other: object) -> bool:
         else:
             return NotImplemented
 
-    def __ne__(self, other: object) -> bool:
-        equality = self.__eq__(other)
-        return not equality if isinstance(equality, bool) else NotImplemented
-
-    def _name_string_as_ordered_set(self) -> str:
-        return "{" + str(self._physical_type_list)[1:-1] + "}"
-
     def __repr__(self) -> str:
         if len(self._physical_type) == 1:
-            names = "'" + self._physical_type_list[0] + "'"
+            names = "'" + self._physical_type[0] + "'"
         else:
-            names = self._name_string_as_ordered_set()
+            names = "{" + str(self._physical_type)[1:-1] + "}"
```

Comment:
If I remember right 🤔, I originally printed out the physical type names as an unordered set, which caused some doctests to fail intermittently since the order was not deterministic. Perhaps we could rename `names` → `ordered_names` to communicate that the order here is important?

Reply:
If you really want to emphasize that `self._physical_type` is sorted then that's what should be renamed.